src/App.js

```diff
@@ -72,6 +73,7 @@ export class ResourceEditor extends React.Component {
     // Here we're only handling single resource but in the future we need to
     // refactor this to manage multiple resources:
     delete resource.sample;
+    resource.id = this.state.resourceId
     datasetMetadata.resources.push(resource);
```

Comment:
I don't think this line is correct. We should find and replace the resource but not push.